doc/overview.rst

```diff
@@ -0,0 +1,13 @@
+Overview
+========
+This section contains system level information about tox.
+
+Environment variables
+---------------------
+tox will inject the following environment variables that you can use to test that your command is running within tox:
+
+.. versionadded:: 3.1
+
+- ``TOX_WORK_DIR`` env var is set to the tox work directory
+- ``TOX_ENV_NAME`` is set to the current running tox environment name
+- ``TOX_ENV_DIR`` is set to the current tox environments working dir.
```

Comment:
should we make these consistent with the names that get substituted in the configuraiton? https://tox.readthedocs.io/en/latest/config.html?highlight=substitution#substitutions

particularly `envname` but `ENV_NAME` seems off to me

should we export all of these substitution parameters as well?

do we even need this feature? someone can `setenv = TOX_ENV_NAME={envname}` already today

Reply:
Although one could do it partially with the ``setenv`` as discussed in the related #847 it adds value to have a general env var name. This way the tools can use to detect if we're inside tox and users no longer need to keep copying around the same set env for their project. 

Coming back to the ``setenv`` I don't think that cuts it. Mostly because setenv is only applied to the current command. I think we want a higher level to be able to be detected by both the package builder and the installer. These are also tools which we invoke; we're not talking here just about commands. In light of this can you have another look?

Reply:
afaik setenv applies to all life cycles (I looked into this for tox-virtualenv-no-download)

Reply:
Hmmm, will double check.

Reply:
Here's that analysis : https://github.com/asottile/tox-virtualenv-no-download/issues/4#issuecomment-362123210